AppCenter/AppCenterTests/MSLogDBStorageTests.m

```diff
@@ -923,11 +983,27 @@ - (void)testMigrationFromSchema2to3 {
   return log;
 }
 
+- (NSArray<id<MSLog>> *)generateAndSaveLogsWithCount:(NSUInteger)count
+                                             groupId:(NSString *)groupId
+                                               flags:(MSFlags)flags
+                              andVerifyLogGeneration:(BOOL)verify {
+  return [self generateAndSaveLogsWithCount:count size:nil groupId:groupId flags:flags storage:self.sut andVerifyLogGeneration:verify];
+}
+
 - (NSArray<id<MSLog>> *)generateAndSaveLogsWithCount:(NSUInteger)count
                                                 size:(NSNumber *)size
                                              groupId:(NSString *)groupId
                                                flags:(MSFlags)flags
-                                             storage:(MSDBStorage *)storage {
+                              andVerifyLogGeneration:(BOOL)verify {
+  return [self generateAndSaveLogsWithCount:count size:size groupId:groupId flags:flags storage:self.sut andVerifyLogGeneration:verify];
+}
+
+- (NSArray<id<MSLog>> *)generateAndSaveLogsWithCount:(NSUInteger)count
+                                                size:(NSNumber *)size
+                                             groupId:(NSString *)groupId
+                                               flags:(MSFlags)flags
+                                             storage:(MSDBStorage *)storage
+                              andVerifyLogGeneration:(BOOL)verify {
```

Comment:
I understand why we do a verification as part of the log creation but the violation of the single responsibility principle makes me uneasy. Still fine as long as it's a conscious decision for a method that's used in tests.

Another question I have is: why do we need the flag? Why don't we verify every log we create?

Reply:
The reason is on the "SELECT" statement below. We need to create logs with random prioirity but SELECT tries to verify all logs stored in the database for specific priority.
For example, if I'd like to generate 2 normal logs, 2 critical logs, and then 2 normal logs, the third generation for 2 normal logs will fail on verification since it expected to create 2 logs but there are 4 logs.

Reply:
I see. That's not obvious, can we add a comment that mentions this?